api/routes/apds/activities/put.js

```diff
@@ -1,58 +1,36 @@
 const logger = require('../../../logger')('apd activites route put');
-const pick = require('lodash.pick');
 const { apdActivity: defaultActivityModel } = require('../../../db').models;
 const {
   loggedIn,
   loadActivity,
+  synchronizeSpecific,
   userCanEditAPD
 } = require('../../../middleware');
 
-module.exports = (app, ActivityModel = defaultActivityModel) => {
-  logger.silly('setting up PUT /activities/:id route');
-  app.put(
-    '/activities/:id',
-    loggedIn,
-    loadActivity(),
-    userCanEditAPD(ActivityModel),
-    async (req, res) => {
-      logger.silly(req, 'handling PUT /activities/:id route');
-      logger.silly(req, `attempting to update activity [${req.params.id}]`);
-
-      try {
-        const activity = req.meta.activity;
-        const activityID = activity.get('id');
-        const apd = req.meta.apd;
-        const newData = pick(req.body, ['name', 'description']);
-
-        if (typeof newData.name !== 'string' || newData.name.length < 1) {
-          logger.verbose(req, 'Invalid activity name');
-          return res
-            .status(400)
-            .send({ error: 'update-activity-invalid-name' })
-            .end();
-        }
-
-        const hasNameConflict = apd
-          .related('activities')
-          .some(
-            a => a.get('id') !== activityID && a.get('name') === newData.name
-          );
+const putters = [null, 'approaches', 'expenses', 'goals', 'schedule'];
 
-        if (hasNameConflict) {
-          logger.verbose(req, 'Activity name already exists for this APD');
-          return res
-            .status(400)
-            .send({ error: 'update-activity-name-exists' })
-            .end();
-        }
+module.exports = (app, ActivityModel = defaultActivityModel) => {
+  putters.forEach(putter => {
+    logger.silly(
+      `setting up PUT /activities/:id${putter ? `/${putter}` : ''} route`
+    );
 
-        activity.set(newData);
-        await activity.save();
-        return res.send(activity.toJSON());
-      } catch (e) {
-        logger.error(req, e);
-        return res.status(500).end();
+    const syncResponder = req => {
+      if (putter) {
+        req.body = { [putter]: req.body };
       }
-    }
-  );
+      return { model: req.meta.activity, action: 'update-activity' };
+    };
+    module.exports.syncResponders[putter || 'base'] = syncResponder;
+
+    app.put(
+      `/activities/:id${putter ? `/${putter}` : ''}`,
+      loggedIn,
+      loadActivity(),
+      userCanEditAPD(ActivityModel),
+      synchronizeSpecific(syncResponder)
+    );
+  });
 };
+
+module.exports.syncResponders = {};
```

Comment:
This module now creates the `put` endpoints for `/activities/:id`, `/activities/:id/approaches`, `/activities/:id/expenses`, `/activities/:id/goals`, and `/activities/:id/schedule`, since they are almost exactly identical.  Just a little bit of conditional logic in this loop and you've got it.